scanner/scanner.go

```diff
@@ -14,7 +14,7 @@ import (
 // ScannerAble is implemented by zbuf.Readers that provide an optimized
 // implementation of the Scanner interface.
 type ScannerAble interface {
-	NewScanner(ctx context.Context, filterExpr ast.BooleanExpr, s nano.Span) (Scanner, error)
+	NewScanner(ctx context.Context, f filter.Filter, expr ast.BooleanExpr, s nano.Span) (Scanner, error)
```

Comment:
s/expr/filterExpr/ everywhere because it isn't obvious that expr ast.BooleanExpr is the filter.